File: frontend/services/analyze.ts
import { apiRequest, endpoints } from "./api";
import type { AnalysisResult } from "./types";

export function analyzeResume(file: File): Promise<AnalysisResult> {
  const formData = new FormData();
  formData.append("file", file);

  return apiRequest<AnalysisResult>(endpoints.analyze, {
    method: "POST",
    body: formData,
  });
}

File: backend/app/routes/analyze.py
from fastapi import APIRouter, File, UploadFile

from app.controllers.analyze_controller import analyze_resume
from app.schemas.resume import AnalysisResult

router = APIRouter(prefix="/api", tags=["analyze"])


@router.post("/analyze", response_model=AnalysisResult)
async def analyze(file: UploadFile = File(...)) -> AnalysisResult:
    return await analyze_resume(file)
